Deployment of py-atlas-building-tools [NSETM-1454]

lguyot opened the pull request:
This change deploys py-atlas-building-tools. 
It is the conclusion of the ticket https://bbpteam.epfl.ch/project/issues/browse/NSETM-1454.

lguyot committed:
Deployment of py-atlas-building-tools [NSETM-1454]

lguyot committed:
Deployment of py-atlas-building-tools [NSETM-1454]

lguyot committed:
Deployment of py-atlas-building-tools [NSETM-1454]

mgeplf commented:
It appears that some of the versions dependencies aren't met:
https://bbpcode.epfl.ch/ci/blue/organizations/jenkins/hpc.spack-deployment.pull-requests/detail/hpc.spack-deployment.pull-requests/1923/tests

Ex: `depends_on('py-networkx@2.5:', type=('build', 'run'))`, but only `2.4` exists in spack at the moment.

lguyot commented:
@mgeplf  @matz-e As Mike pointed out, the tests are failing because of unmet requirements for:
- py-rtree
- py-numpy-quaternion
- py-networkx
- py-openpyxl
- py-morphio (BBP)
The immediate action that I can take is to add for each of the above spack packages more recent versions in their package.py files. Would this be OK?

mgeplf commented:
Yup, that's exactly what needs to be done.

I've heard that @matz-e likes it if the same patches go to upstream, so maybe he can help to make that smooth?  I believe it saves pain down the road.

mgeplf commented:
@lguyot: I should note that the other option is to check if the older versions work: perhaps `networkx-2.4` is fine for the `py-atlas-building-tools`, and its setup.py:requirements could be loosened to use `networkx-2.4`.

lguyot commented:
@mgeplf I have made tests, lowering requirement versions so that they match what is available in BBP spack.
I can lower successfully almost everything:
- py-rtree
- py-networkx
- py-openpyxl
- py-trimesh (not reported as a failure, but it could, since py-atlas-building-tools requires py-trimesh>=3.6.18)

Still, for py-numpy-quaternion, I have no choice: I need to specify in py-numpy-quaternion/package.py an older version, namely 2019.12.11.22.25.52, which is required because of an incompatibility with numpy==1.19.4 and older versions.

For the py-morphio failing test, this is a puzzle: py-atlas-building-tools doesn't seem to have any dependency on it.

Eventually, lowering py-trimesh>=3.6.18 (current requirement in py-atlas-building-tools and atlas-building-tools) to py-trimesh>=2.38.10 (available spack version) will be possible only if cgal-pybind does lower its dependency version too, since atlas-building-tools depends on cgal-pybind.

Here is my plan:
- (1) let review and merge https://bbpcode.epfl.ch/code/#/c/53354/1/setup.py (lower atlas-building-tools requirements)
- (2) release atlas-building-tools 0.1.1
- (3) back to this PR: lower the requirements in py-atlas-building-tools

 It is possible however that something needs to be done for cgal-pybind before performing step (1) above:
- (a) lower the trimesh requirement of cgal-pybind;
- (b) release cgal-pybind 0.1.0
- (c) update py-cgal-pybind/package.py with the new version and the trimesh requirement.

GianlucaFicarelli committed:
Bump workflow (py-morphology-repair-workflow 2.0.3) (#1099)

Bump workflow

Bump py-morphology-repair-workflow and update related packages.
Keep only the latest py-morphio/morphio 3.1.1.

GianlucaFicarelli commented:
Commenting just about py-morphio: it should be fixed if you rebase on the current `develop` branch because #1099 has been merged.

lguyot commented:
@GianlucaFicarelli Thanks! I'll do that soon.

NadirRoGue committed:
Add Brion 3.3.2 release (#1155)

* Add Brion 3.3.2 release

* Update var/spack/repos/builtin/packages/brion/package.py

Co-authored-by: Matthias Wolf <m+git@sushinara.net>

Co-authored-by: Matthias Wolf <m+git@sushinara.net>

arnaudon committed:
Region grower v2 (#1153)

* region-grower v2

* update depends_on + neurom

* Update var/spack/repos/builtin/packages/py-diameter-synthesis/package.py

* fix depends_on

* fix neurom

* Update var/spack/repos/builtin/packages/py-tns/package.py

* Update var/spack/repos/builtin/packages/py-tns/package.py

* Update var/spack/repos/builtin/packages/py-tns/package.py

* Update var/spack/repos/builtin/packages/py-region-grower/package.py

* Update var/spack/repos/builtin/packages/py-region-grower/package.py

* added morphio in diameter-synthesis;

* dask + attrs

* fix

alexsavulescu committed:
Neurodamus-py 2.6.0 (#1157)

* py-morphio is as of now a run dependency

olupton committed:
Bump NEURODAMUS/NEURON/CoreNEURON/NMODL/LIBSONATA-REPORT versions. (#1146)

* Add neuron v8.0.0, tweak py-pytest-cov dependency.
* Add new tags of NMODL and CoreNEURON.
* Rename nmodl 0.3 to 0.3.0.
* Add new tags for neurodamus-core and libsonata-report

Co-authored-by: Jorge Blanco Alonso <jorge.blancoalonso@epfl.ch>
Co-authored-by: Ioannis Magkanaris <iomagkanaris@gmail.com>

mgeplf committed:
have emsim depend on brion 3.3.2 (#1158)

lguyot committed:
Add version 0.1.0 to py-cgal-pybind/package.py [NSETM-1454] (#1159)

* Add version 0.1.0 to py-cgal-pybind/package.py [NSETM-1454]

This new version has no install dependency anymore on trimesh
but only an extra dependency named "tests".

This allows consumers of the cgal-pybind API like atlas-building-tools
to lower the required version of trimesh so that it matches the versions
available in BBP's py-trimesh/package.py (currently trimesh==2.38.10).

See https://github.com/BlueBrain/spack/pull/1152 for full context.

* Fixes flake8 errors

* Puts latest version at the top of the version list (package.py)

* Refers to the gitlab repository only

* Removes older version 0.0.2: it is actually unused

jorblancoa committed:
libsonata-report: Add version 1.0 (#1160)

lguyot committed:
Lower the requirements on the dependencies of py-atlas-building-tools to match the available versions of BBP spack

lguyot committed:
Lower the requirements on the dependencies of py-atlas-building-tools to match the available versions of BBP spack

lguyot committed:
Lower the requirements on the dependencies of py-atlas-building-tools to match the available versions of BBP spack

GianlucaFicarelli commented on var/spack/repos/builtin/packages/py-atlas-building-tools/package.py:
```diff
@@ -11,31 +11,31 @@ class PyAtlasBuildingTools(PythonPackage):
     depends_on('py-setuptools', type=('build', 'run'))
 
     depends_on('py-cgal-pybind', type=('build', 'run'))
     depends_on('py-click@3.0:', type=('build', 'run'))
```
```suggestion
    depends_on('py-click@7.0:', type=('build', 'run'))
```

GianlucaFicarelli commented on var/spack/repos/builtin/packages/py-atlas-building-tools/package.py:
```diff
@@ -11,31 +11,31 @@ class PyAtlasBuildingTools(PythonPackage):
 
     depends_on('py-setuptools', type=('build', 'run'))
 
     depends_on('py-cgal-pybind', type=('build', 'run'))
```
```suggestion
    depends_on('py-cgal-pybind@0.1.0:', type=('build', 'run'))
```

GianlucaFicarelli commented on var/spack/repos/builtin/packages/py-atlas-building-tools/package.py:
```diff
@@ -11,31 +11,31 @@ class PyAtlasBuildingTools(PythonPackage):
-    depends_on('py-openpyxl@3.0.5:', type=('build', 'run'))
+    depends_on('py-openpyxl@3.0.3:', type=('build', 'run'))
     depends_on('py-pandas@1.0.3:', type=('build', 'run'))
     depends_on('py-pillow@1.7.2:', type=('build', 'run'))
```
```suggestion
    depends_on('py-pillow@7.1.2:', type=('build', 'run'))
```

GianlucaFicarelli commented on var/spack/repos/builtin/packages/py-atlas-building-tools/package.py:
```diff
@@ -11,31 +11,31 @@ class PyAtlasBuildingTools(PythonPackage):
+    depends_on('py-openpyxl@3.0.3:', type=('build', 'run'))
     depends_on('py-pandas@1.0.3:', type=('build', 'run'))
     depends_on('py-pillow@1.7.2:', type=('build', 'run'))
     depends_on('py-poisson-recon-pybind', type=('build', 'run'))
```
```suggestion
    depends_on('py-poisson-recon-pybind@0.1.0:', type=('build', 'run'))
```

GianlucaFicarelli commented on var/spack/repos/builtin/packages/py-atlas-building-tools/package.py:
```diff
@@ -11,31 +11,31 @@ class PyAtlasBuildingTools(PythonPackage):
     git      = "ssh://bbpcode.epfl.ch/nse/atlas-building-tools"
 
-    version('0.1.0', commit='4d2ca679e34d08c76f96f15ccd3cdeccd2044696')
+    version('0.1.1', tag='atlas_building_tools-v0.1.1')
```
Not for this PR, but it seems that also `atlas_building_tools` uses `_` instead of `-` as separator for `name` in `setup.py` (as it was for `cgal-pybind`).

lguyot replied:
@GianlucaFicarelli  Thanks for all your comments and corrections.
Regarding the name used in setup.py, we have indeed the same problem as before.

I am in the process of writing a new commit in atlas-building-tools. The first thing I did was to change the name in setup.py, using dashes instead of underscores.

So it will be fixed soon.

lguyot committed:
Apply suggestions from code review

Co-authored-by: Gianluca Ficarelli <26835404+GianlucaFicarelli@users.noreply.github.com>

lguyot committed:
Apply suggestions from code review

Co-authored-by: Gianluca Ficarelli <26835404+GianlucaFicarelli@users.noreply.github.com>

lguyot committed:
Apply suggestions from code review

Co-authored-by: Gianluca Ficarelli <26835404+GianlucaFicarelli@users.noreply.github.com>

lguyot committed:
Apply suggestions from code review

Co-authored-by: Gianluca Ficarelli <26835404+GianlucaFicarelli@users.noreply.github.com>

lguyot committed:
Apply suggestions from code review

Co-authored-by: Gianluca Ficarelli <26835404+GianlucaFicarelli@users.noreply.github.com>

lguyot committed:
Apply suggestions from code review

Co-authored-by: Gianluca Ficarelli <26835404+GianlucaFicarelli@users.noreply.github.com>

lguyot committed:
Indicates that py-atlas-building-tools is a BBP package in package.py

lguyot committed:
Indicates that py-atlas-building-tools is a BBP package in package.py

lguyot committed:
Indicates that py-atlas-building-tools is a BBP package in package.py

matz-e committed:
brion: add glue for dependent specs (#1161)

Commit:
highfive: version bump to 2.3, drag synapsetools along (#1151)

Co-authored-by: Matthias Wolf <matthias.wolf@epfl.ch>

matz-e committed:
Changed stages permissions to 750 (#21425) (#1163)

Co-authored-by: BenWeber42 <dev.ben.weber@gmail.com>

arnaudon committed:
added to whitelist (#1164)

sergiorg-hpc committed:
Adds support for Caliper variant in CoreNEURON (#1156)

* Adds support for Caliper variant and enables it by default.
* Updates Caliper recipe from upstream and fixes an issue in CoreNEURON.

olupton committed:
build tests: put an upper bound on the GCC version (#1168)

Backport of https://github.com/spack/spack/pull/23630, original message:
 - build tests: put an upper bound on the version of GCC being used

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

olupton committed:
Add nvhpc/pgi localrc files for *all* nodes. (#1166)

sergiorg-hpc committed:
Update CoreNEURON package to be consistent with the CI (#1167)

* Updates the CoreNEURON package to be consistent with the CI
* Clean references and configurations to old versions

Commit:
steps: add py-mpi4py as dep (#1169)

lguyot committed:
Merge branch 'py-atlas-building-tools-deployment' of https://github.com/BlueBrain/spack into py-atlas-building-tools-deployment

lguyot committed:
Revert "Indicates that py-atlas-building-tools is a BBP package in package.py"

This reverts commit af6f2385a138961e3eece82656c21e9cf73a43d8.

lguyot committed:
Revert "Indicates that py-atlas-building-tools is a BBP package in package.py"

This reverts commit af6f2385a138961e3eece82656c21e9cf73a43d8.

lguyot committed:
Revert "Merge branch 'py-atlas-building-tools-deployment' of https://github.com/BlueBrain/spack into py-atlas-building-tools-deployment"

This reverts commit 9a1e6ea0a06c683d833d0f9ad259023f1872cb0a, reversing
changes made to b119f4869fc22400b0c1f3e3c61d1ad1e05c682c.

lguyot committed:
Revert "Merge branch 'py-atlas-building-tools-deployment' of https://github.com/BlueBrain/spack into py-atlas-building-tools-deployment"

This reverts commit 9a1e6ea0a06c683d833d0f9ad259023f1872cb0a, reversing
changes made to b119f4869fc22400b0c1f3e3c61d1ad1e05c682c.

lguyot committed:
Merge branch 'py-atlas-building-tools-deployment' of https://github.com/BlueBrain/spack into py-atlas-building-tools-deployment

lguyot commented:
I have been seriously messing up with the git  rebase / merge / revert operations. I am going to close this pull-request and re-start from a clean slate.
